src/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+
+require("./TextInput.less");
+
+export class TextInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {inFocus: false};
+    this.onFocus = this.onFocus.bind(this);
+    this.onBlur = this.onBlur.bind(this);
```

Comment:
optional: you don't need to do the binding if you use arrow functions like this: 
```
onFocus = () => { 
  ...
}
```